simpleclient_common/src/main/java/io/prometheus/client/exporter/common/TextFormat.java

```diff
@@ -53,6 +124,10 @@ public static void write004(Writer writer, Enumeration<Collector.MetricFamilySam
         writer.write('\n');
       }
     }
+    // Write out any OM-specific samples.
+    if (!omFamilies.isEmpty()) {
+      write004(writer, Collections.enumeration(omFamilies.values()));
```

Comment:
Hm, could this not be recursive since this is in `write004` method itself? i.e. could it not each time collect into `omFamilies` and recurse but not actually writing it in the child recursive call but simply recursing again infinitely?

Haven't looked to see if there's a test or not for this just yet.

Reply:
The `if` is there for a reason. Java lacks inner functions, so this is the easy way to not repeat the rendering logic.

Reply:
Ah yes, I see - and these will be gauges in the recurse function. Makes sense.